### DISCUSSION TITLE
The scripts in the repository requires Proxmox Virtual Environment 8.1 or later.

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#2433] [Announcements]
Scripts will not execute on PVE7. Subsequent <a href='https://forum.proxmox.com/threads/proxmox-ve-support-lifecycle.35755/' target='_blank' rel='noopener noreferrer'>Proxmox VE - Support Lifecycle</a>

**Comment by tteck** [comment]
Starting today (2024-02-06), the scripts will **only** work with PVE7 Version **7.4-13** or later, or PVE8 Version **8.1.1** or later.

**Reply by crankshaft255** [reply]
I get this error when trying to use scripts now.  

 ✗ This version of Proxmox Virtual Environment is not supported
Requires PVE Version 7.4-13 or higher

but this is what I am running
```
root@prox:~# pveversion
pve-manager/8.0.3/bbf3993334bfa916 (running kernel: 6.2.16-3-pve)
root@prox:~# 
```

**Reply by tteck** [reply]
I will update the language to also include 8.1.1 or higher.

```
if ! pveversion | grep -Eq "pve-manager/(7.4-[1][3-9]|8.1.[1-9])"; then
    msg_error "This version of Proxmox Virtual Environment is not supported"
```

**Reply by tteck** [reply]
Requires PVE7 Version 7.4-13 or later, or PVE8 Version 8.1.1 or later.

**Reply by crankshaft255** [reply]
I ran a dist upgrade and my version in only 8.0.3

**Reply by tteck** [reply]
The latest is `pve-manager/8.1.4/ec5affc9e41f1d79 (running kernel: 6.5.11-8-pve)`

**Reply by theminkkilledcuriosity** [reply]
Im getting the same error after a fresh install from 8.2 ISO


This is what is running after a fresh install and running post install script:

pve-manager/8.2.2/9355359cd7afbae4 (running kernel: 6.8.4-3-pve)

**Comment by tteck** [comment]
> I ran a dist upgrade and my version in only 8.0.3

You're still on Beta.

**Reply by crankshaft255** [reply]
Yeah my mistake. I found the right repositories to add to upgrade to the latest and scripts are working like a charm

**Reply by theminkkilledcuriosity** [reply]
what repos are you using that worked?

**Reply by tteck** [reply]
What version are you on now?

**Reply by theminkkilledcuriosity** [reply]
This is what is running after a fresh install and running post install script:

pve-manager/8.2.2/9355359cd7afbae4 (running kernel: 6.8.4-3-pve)

**Reply by tteck** [reply]
Then you're good.

**Reply by theminkkilledcuriosity** [reply]
is 6.8.4-3-pve a correct kernel? I was under the impression it wasnt updating because of running pve7to8 I get: 

Checking running kernel version..
WARN: unexpected running and installed kernel '6.8.4-3-pve'.

and using the created pve8 script I get:
 
This version of Proxmox Virtual Environment is not supported
  PVE Version 7.4-16 or higher is required.


So I was under the impression I was not upgrading/upgraded. but if Im good then I can finally relax lmao.

**Comment by ither** [comment]
Can you please explain why the need to support ony 8.1.1 and up? I have 8.0.4 installed and I can't upgrade right now. It is for simplicity sake, I presume. Thanks for all your wonderful work.

**Reply by tteck** [reply]
> I have 8.0.4 installed and I can't upgrade right now.

That's the beta version. So, currently you get the "`✗ This version of Proxmox Virtual Environment is not supported`", correct?
Also, why wouldn't you be able to upgrade?

**Reply by ither** [reply]
Huh, I didn't know that 8.0.4 was a beta version. Some time ago I upgraded from version 7 and a network driver gave me serious headaches. Since then, for upgrading Proxmox I want to be in the physical location of the server, which at the current time I access remotely until the next month. 

Now make sense that the network driver gave me troubles. I didn't see any warnings that the version 8.0.x was beta, may be I just overlooked it. I need to pay more attention.

Anyway, I don't feel confident in upgrading remotely now. Do you think it should work? Don't worry, I won't make you responsible if anythin happens.

**Comment by SofieF2005** [comment]
When I run 

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/mqtt.sh)"

in shell nothing will happen. Why is that?

Im on Proxmox 8.1.4

**Reply by tteck** [reply]
It seems that you're facing a network problem, _probably_ linked to DNS.

**Reply by SofieF2005** [reply]
yes. U are right.

DNS was 127.0.0.1  ;)

**Comment by uncharted9898** [comment]
Should have created a v7 fork before you "broke" the repo for some. I guess that's what snapshots are for!

For any interested, last commit before the version force: https://github.com/tteck/Proxmox/tree/65cf479b99deae11aed4e8fee482c19784790010

**Reply by tteck** [reply]
Considering that Proxmox will cease support for PVE7 in July, why not upgrade to the latest Proxmox version and enjoy the benefits?

**Reply by uncharted9898** [reply]
I was fine with the hoops of version install until today, when I tried to install another vm and found you removed that option. 

Ever heard of the ole saying " If it ain't broke don't fix it " 

I feel it applies here, EoL aside. I'm all about options and you removed em. It is what it is,

**Reply by tteck** [reply]
Whether you choose to fix your system to be compatible with these scripts or not, I really don't care. However, advancements in future scripts will be achievable without backward compatibility being a concern.

Here's another ole saying "Adapt or get left behind"

**Reply by marksie1988** [reply]
v8 has some major issues with the new kernel 6.8.x, hence why i cant upgrade, it bricked a few of my servers and they didnt work properly after. forcing people to use it is a rubbish move when its easy to tag the old version and let people use it

**Reply by tteck** [reply]
I'm sorry to hear about the issues you've experienced with kernel 6.8.x. My experience with the newer kernels has been stable, but I understand that may not be the case for everyone.

I am not forcing anyone to use Proxmox VE or these scripts. However, starting in July 2024, the scripts in the repository will require Proxmox Virtual Environment 8.1 or later due to the new features and improvements this version (and future versions) provides. I understand that backward compatibility is a concern, but maintaining compatibility with older versions is hindering the process of script improvement moving forward (e.g., Ubuntu 24.04).

If sticking to an older version is necessary for your setup, you might consider creating a fork of the repository and maintaining it for the Proxmox VE version you're using. This way, you can ensure compatibility with your environment.

**Reply by marksie1988** [reply]
No need to maintain backwards compatibility, just have a version tag for the scripts that work with 7.4 so it's easy to use them.

**Reply by tteck** [reply]
Currently, you receive a message stating, "_You are currently using Proxmox VE 7 (EOL 2024-07), refrain from creating Debian 12 LXCs_," most of the scripts default to Debian 12. In reality, very few scripts would have a 7.4 tag. I'd suggest working out the issues with Proxmox VE 8 on your servers.

**Reply by marksie1988** [reply]
No I don't think you understand what I meant, put a GitHub tag that is at the version of scripts before the change to only supporting 8. That way people can click the tag in GitHub and be at the scripts that work with 7.4

**Reply by tteck** [reply]
> If sticking to an older version is necessary for your setup, you might consider creating a fork of the repository and maintaining it for the Proxmox VE version you're using. This way, you can ensure compatibility with your environment.

I will no longer be providing support for PVE 7 come July.

**Reply by marksie1988** [reply]
> > If sticking to an older version is necessary for your setup, you might consider creating a fork of the repository and maintaining it for the Proxmox VE version you're using. This way, you can ensure compatibility with your environment.
> 
> I will no longer be providing support for PVE 7 come July.

Ok I'm not asking for support, merely a GitHub tag to help find the old version scripts that work

**Comment by erikvanberkum** [comment]
Petty we are still on 7 as on 8 numa support is not reliable and for this reason we are not upgrading yet.

https://forum.proxmox.com/threads/proxmox-8-0-kernel-6-2-x-100-cpu-issue-with-windows-server-2019-vms.130727/page-9

**Reply by tteck** [reply]
Hopefully, by July, they will find a solution instead of relying on a workaround.

**Reply by tteck** [reply]
Maybe 🤞🏻 
```
proxmox-kernel-6.5 (6.5.13-1) bookworm; urgency=medium

  * update sources to Ubuntu-6.5.0-20.20 based on v6.5.13 and some newer
    stable-tree backports

  * backport scheduler fix to avoid temporary VM freezes on NUMA hosts
  ```

**Comment by crankshaft255** [comment]
I just ran another update and have this version
proxmox-ve: 8.1.0 (running kernel: 6.5.11-8-pve)
pve-manager: 8.1.4 (running version: 8.1.4/ec5affc9e41f1d79)

However when I try to run a script I get this

⚠️  Can't Install on Proxmox

**Reply by tteck** [reply]
This is because you're trying to execute a script on the Proxmox host that is not allowed. Read the instruction for that particular script.

**Comment by crankshaft255** [comment]
My bad, really not enough coffee in me this morning yet.  I totally skipped
over the part where it said to do it in the LXC container.

Thank you for all the work you do :)
